### ISSUE TITLE
dotnet nuget restore is not caching the packages in linux

### ISSUE THREAD

**Issue by dei79** [Microsoft/vsts-tasks#5004]
Currently I'm using the dotnet core tasks to restore the packages for my project. This process takes 3 up to 5 minutes which just delays my builds. It looks like the dotnet core tasks are not solving the caching problem otherwise the restore task should be finished in ~10 seconds like it happens on my Mac when nothing was changed. 

Can you advice me how I can ensure that the nuget packages are globally cached as described in NuGet 4.X?

**Comment by bryanmacfarlane** [comment]
Which queue / pool are you using?

**Comment by dei79** [comment]
It's a private queue based on the official VSTS Docker Agent on Ubuntu Linux

**Comment by ds-ms** [comment]
@dei79 What is the version of the task you're using? 
Also, is it possible that the build is running on a different machine altogether when this happens?

**Comment by dei79** [comment]
I'm using the following 

2017-08-09T22:17:22.3577440Z ==============================================================================
2017-08-09T22:17:22.3600040Z Task         : .NET Core
2017-08-09T22:17:22.3665130Z Description  : Build, test and publish using dotnet core command-line.
2017-08-09T22:17:22.3694240Z Version      : 1.0.2
2017-08-09T22:17:22.3715180Z Author       : Microsoft Corporation
2017-08-09T22:17:22.3756400Z Help         : [More Information](https://go.microsoft.com/fwlink/?linkid=832194)
2017-08-09T22:17:22.3789200Z ==============================================================================

I just have one container for testing so every build runs on the same agent. 

Where can I see that the caching is used or not? Because I want to ensure that it is not just the machine which is slow?

**Comment by ds-ms** [comment]
If you see the logs, you can find the exact command which is being executed. We don't disable caching. The default behavior of dotnet-core is maintained. 

[Here's](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-restore) some documentation about the restore command. The description has information about where the nuget packages are restored to in different environments. 

PS: To avoid caching, you'll have to explicitly use --no-cache argument when running the command

**Comment by dei79** [comment]
So what I can say now after reading logs and understanding more. I used the task with the pattern **/*.csproj which means a nuget restore was executed for all projects in my solution. I learned then that nuget restore for my master project is restoring also all packages for the dependencies. After changing the pattern to a dedicate project it's clear that the caching is working. Now the restore just takes 5 up to 10 seconds as expected.
